src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+    let newSeperator = separator;
+
+    if (separator === null) {
+      newSeperator = 'null';
+    }
+
+    for (const key of this) {
+      switch (key) {
+        case null:
+        case undefined:
+          result += newSeperator + '';
+          break;
+        default:
+          result += newSeperator + key;
+          break;
+      }
```

Comment:
you don't seem to need the whole switch
```suggestion
        result += newSeperator + key;
    }
```